lib/gui/models/store.js

```diff
@@ -287,10 +287,22 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         throw errors.createError('Missing image size');
       }
 
-      if (!_.isNumber(action.data.size)) {
+      if (!_.isPlainObject(action.data.size)) {
         throw errors.createError(`Invalid image size: ${action.data.size}`);
       }
 
+      if (!_.isNumber(action.data.size.original)) {
```

Comment:
I'm assuming you meant declaratively rather than decoratively ;-)

Sounds good, as long as parsing and validating the schema doesn't slow things down.
Hmm, I wonder if we could get the best of both worlds and use some module which (at build time - like we do with the sass) converts a schema into a list of conditionals to test? :)